---
title: C26110
ms.date: 10/01/2019
ms.topic: reference
f1_keywords:
- C26110
helpviewer_keywords:
- C26110
ms.assetid: d82b79ec-6d7f-438b-bd6a-da874a3e08e5
author: mikeblome
ms.author: mblome
manager: markl
ms.workload:
- multiple
ms.openlocfilehash: 6f72dc7dbd9941c36bdf35c3a2c88541e854e393
ms.sourcegitcommit: 485ffaedb1ade71490f11cf05962add1718945cc
ms.translationtype: MT
ms.contentlocale: ru-RU
ms.lasthandoff: 10/16/2019
ms.locfileid: "72435149"
---
# <a name="c26110"></a>C26110
предупреждение C26110: вызывающему объекту не удалось сохранить \<lock блокировки > перед вызовом функции \<func >.

 Если требуется блокировка, обязательно выясните, должна ли сама функция или ее вызывающая сторона получить блокировку. Предупреждение C26110 выдается при нарушении `_Requires_lock_held_` аннотации или других заметок, связанных с блокировкой. Дополнительные сведения см. в разделе [Аннотирование режима блокировки](annotating-locking-behavior.md) .

## <a name="example"></a>Пример
 В следующем примере создается предупреждение C26110, так как Аннотация `_Requires_lock_held_` on Function `LockRequired` указывает, что вызывающий объект `LockRequired` должен получить блокировку перед вызовом `LockRequired`. Без этой заметки `LockRequired` должен получить блокировку до того, как она будет обращаться к любым общим данным, защищенным блокировкой.

```

typedef struct _DATA
{
    CRITICAL_SECTION cs;
    int d;
} DATA;

_Requires_lock_held_(p->cs)

void LockRequired(DATA* p)
{
    p->d = 0;
}

void LockNotHeld(DATA* p)
{
    LockRequired(p); // Warning C26110
}
```
